hbase-replication/src/main/java/org/apache/hadoop/hbase/replication/ReplicationTrackerZKImpl.java

```diff
@@ -168,7 +169,7 @@ private boolean refreshOtherRegionServersList(boolean watch) {
     } else {
       synchronized (otherRegionServers) {
         otherRegionServers.clear();
-        otherRegionServers.addAll(newRsList);
+        newRsList.stream().map(ServerName::parseServerName).forEach(otherRegionServers::add);
```

Comment:
Nit: Maybe worth do this conversion inside `getRegisteredRegionServers` method, instead, as it's the first point of entry from ZK where data comes as Sting still? Just in case we may find useful use `getRegisteredRegionServers` method elsewhere, than it would already be returning a list of `ServerName`.

Reply:
Thank @wchevreuil for reviwing, your opinion is indeed better, I have revised it.